### PR TITLE
[TestNG] Log method parameters in tests

### PR DESCRIPTION
**Pull request by jiangty-addepar**
Replacing #1664 now that the other PRs have been merged.

TestNG has features like `@DataProvider` which involve passing in parameters to tests. This changes Buck to log those test parameters.

Test Plan: `TestNGLoggingTest`

### CODE REVIEW

File: src/com/facebook/buck/testrunner/TestNGRunner.java [hunk 1]
```diff
@@ -55,9 +63,18 @@ public void run() throws Throwable {
       } else {
         results = new ArrayList<>();
         TestNG testng = new TestNG();
+        testng.setUseDefaultListeners(false);
         testng.setAnnotationTransformer(new FilteringAnnotationTransformer(results));
-        testng.setTestClasses(new Class<?>[] {testClass});
```
Comment (ttsugriy) [inline src/com/facebook/buck/testrunner/TestNGRunner.java]: looks like you're breaking formatting here... Please use google-format to fix all formatting issues

Comment (jiangty-addepar) [reply src/com/facebook/buck/testrunner/TestNGRunner.java]: I installed the IntelliJ plugin, but I guess it doesn't auto-format :(

**Comment by jiangty-addepar** [comment]
@ttsugriy sorry about that!

Is there an easy way to tell IntelliJ to run formatting on all changed files or something?

**Comment by ttsugriy** [comment]
sorry, I don't know @jiangty-addepar, we have `google-format` tool integrated with arcanist tool so it happens as part of `arc lint` workflow :(

**Comment by ttsugriy** [comment]
@jiangty-addepar, looks like your changes break the build - https://ci.appveyor.com/project/Facebook/buck/build/5912#L2252

**Comment by jiangty-addepar** [comment]
I'm really confused -- I'm running this test locally and it not only compiles, but passes?

**Comment by jiangty-addepar** [comment]
hmm, perhaps it's a semantic merge conflict; I'll rebase on master and try again

File: src/com/facebook/buck/testrunner/TestNGRunner.java [hunk 2]
```diff
@@ -114,6 +131,30 @@ private boolean shouldIncludeTest(String className) {
...
+      return name;
+    }
+
+    StringBuilder builder = new StringBuilder(name).append(" (");
+    builder.append(
+        Arrays.stream(parameters)
+            .map(
+                parameter -> {
+                  try {
+                    return parameter == null ? "null" : parameter.toString();
```
Comment (styurin) [inline src/com/facebook/buck/testrunner/TestNGRunner.java]:
```
return String.valueOf(parameter);
```

File: src/com/facebook/buck/testrunner/TestNGRunner.java [hunk 3]
```diff
@@ -114,6 +131,30 @@ private boolean shouldIncludeTest(String className) {
...
+                  } catch (Exception e) {
+                    return "Unstringable object";
```
Comment (styurin) [inline src/com/facebook/buck/testrunner/TestNGRunner.java]: Why do we need to catch it and not fail fast?

File: src/com/facebook/buck/testrunner/TestNGRunner.java [hunk 4]
```diff
@@ -247,4 +289,11 @@ private PrintStream streamToPrintStream(ByteArrayOutputStream str, PrintStream f
       }
     }
   }
+
+  private static class JUnitReportReporterImproved extends JUnitReportReporter {
```
Comment (styurin) [inline src/com/facebook/buck/testrunner/TestNGRunner.java]: `Improved` is not a great term to use. Could you replace it with a name that explain the purpose of this class.

File: src/com/facebook/buck/testrunner/TestNGRunner.java (hunk 3 above)
Comment (jiangty-addepar) [reply src/com/facebook/buck/testrunner/TestNGRunner.java]: we can fail instead, but I don't think "failure to `toString` an object" should be enough to block a developer from seeing their test results

**Pull request closed without merging**